coordinator/src/main/scala/filodb.coordinator/queryplanner/LongTimeRangeSplitPlanner.scala

```diff
@@ -0,0 +1,118 @@
+package filodb.coordinator.queryplanner
+
+import com.typesafe.scalalogging.StrictLogging
+import debox.Buffer
+
+import filodb.coordinator.queryplanner.LogicalPlanUtils._
+import filodb.core.query.QueryContext
+import filodb.query.{LogicalPlan, PeriodicSeriesPlan}
+import filodb.query.exec.{ExecPlan, PlanDispatcher, StitchRvsExec}
+
+/**
+  * LongTimeRangeSplitPlanner splits query along the time dimension, based on a configured threshold (splitThresholdMs).
+  * It knows about limited retention of raw data, and existence of downsampled data.
+  * For any query that arrives beyond the retention period of raw data, it splits the query into
+  * two time ranges - latest time range from raw data, and old time range from downsampled data.
+  * It then splits each of the query ranges into multiple smaller ranges (splitSizeMs).
+  * And finally stitches the subquery results to present top level query results for entire time range.
+  *
+  * @param rawClusterPlanner this planner (typically a SingleClusterPlanner) abstracts planning for raw cluster data
+  * @param downsampleClusterPlanner this planner (typically a SingleClusterPlanner)
+  *                                 abstracts planning for downsample cluster data
+  * @param earliestRawTimestampFn the function that will provide millis timestamp of earliest sample that
+  *                             would be available in the raw cluster
+  * @param latestDownsampleTimestampFn the function that will provide millis timestamp of newest sample
+  *                                    that would be available in the downsample cluster. This typically
+  *                                    is not "now" because of the delay in population of downsampled data
+  *                                    via spark job. If job is run every 6 hours,
+  *                                    `(System.currentTimeMillis - 12.hours.toMillis)`
+  *                                    may a function that could be passed. 12 hours to account for failures/reruns etc.
+  * @param splitThresholdMs the function that will provide the threshold in milliseconds. If the query time ranges is
+  *                         beyond this threshold, then the query will be split into smaller time range queries.
+  * @param splitSizeMs  If the query is to be broken then it will be of this size
+  * @param stitchDispatcher function to get the dispatcher for the stitch exec plan node
+  */
+class LongTimeRangeSplitPlanner(rawClusterPlanner: QueryPlanner,
```

Comment:
As mentioned, I don't think we need another planner. The contents can be folded in as a helper class to existing planner.

Reply:
@vishramachandran moved logic inside SingleClusterPlanner and removed this new planner.